### PR TITLE
Balance display BAT and USD

### LINKED ISSUE #155: Update balance to use BAT
1. /v1/publishers/#{publisher.brave_publisher_id}/balance

if you call "/v2/..." instead, you'll get back something like this:

```json
{
  "amount": "5.80",
  "currency": "USD",
  "altcurrency": "BAT",
  "probi": "25000000000000000000",
  "rates": {
    "BTC": 0.00005418424016883016,
    "ETH": 0.000795331082073117,
    "USD": 0.2363863335301452,
    "EUR": 0.20187818378874756,
    "GBP": 0.1799810085548496
  }
}
```

the balance display should be like this:

```html
   25 BAT (5.80 USD)
```

(at least, that's what i am told today...)

to go from probi to BAT you do this (in node)

```javascript
  new BigNumber(probi.toString()).dividedBy(1e18).toFormat(2)
```

you can pretty much ignore the rates for now.

2. also, can you change the existing call of

```      
   PUT /v1/publishers/#{publisher.brave_publisher_id}
```

to

```
   PUT /v1/publishers/#{publisher.brave_publisher_id}/wallet
```

Per @mrose17 in Slack

### PR DESCRIPTION
**Pull request by lgebhardt**
- Adds support for the new `/v2/publishers/#{publisher.brave_publisher_id}/balance` endpoint
- Adds a new Eyeshade::Balance class to handle the new response format
- Updates the dashboard to use the new formatted balance.
- Updates the wallet setter endpoint

Fixes #155

### CODE REVIEW

File: app/helpers/publishers_helper.rb [hunk 1]
```diff
@@ -24,13 +24,16 @@ def poll_uphold_status?(publisher)
...
+  # balance: Instance of Eyeshade::Balance
   def publisher_humanize_balance(publisher)
     if balance = publisher.balance
-      number_to_currency(balance.amount)
+      "#{'%.2f' % balance.BAT} BAT (#{'%.2f' % balance.convert_to('USD')} USD)"
     else
       I18n.t("publishers.balance_error")
     end
+  rescue => e
+    Rails.logger.warn(e)
```
Comment (ayumi) [inline app/helpers/publishers_helper.rb]: let's add a `Raven.capture_exception(e)` for extra API logging

File: app/services/publisher_balance_getter.rb [hunk 2]
```diff
@@ -8,26 +10,35 @@ def initialize(publisher:)
...
   rescue Faraday::Error => e
     Rails.logger.warn("PublisherBalanceGetter #perform error: #{e}")
     nil
   end
 
   def perform_offline
     Rails.logger.info("PublisherVerifier returning offline stub balance.")
-    Balance.new("9001.00", "USD", 150000000)
+    Eyeshade::Balance.new(
+        balance_json: {
```
Comment (ayumi) [inline app/services/publisher_balance_getter.rb]: indentation: 4 spaces instead of 2

File: lib/eyeshade/balance.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      @probi / BigDecimal.new('1.0e18')
+    end
+
+    def convert_to(currency_code = 'USD')
+      rate = rates[currency_code]
+      raise "Missing currency conversion rate #{currency_code} for #{@balance_json}" unless rate
+
+      self.BAT * rate
+    end
+
```
Comment (ayumi) [inline lib/eyeshade/balance.rb]: minor: trailing newline

File: test/services/publisher_balance_getter_test.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    Rails.application.secrets[:api_eyeshade_offline] = prev_api_eyeshade_offline
+  end
+
+  test "when online gets a balance" do
+    prev_api_eyeshade_offline = Rails.application.secrets[:api_eyeshade_offline]
+    Rails.application.secrets[:api_eyeshade_offline] = false
+
+    balance_json = {
+        "amount" => "5.80",
```
Comment (ayumi) [inline test/services/publisher_balance_getter_test.rb]: minor: 4 space indentation vs 2

**Comment by mrose17** [comment]
@lgebhardt ... will you be adding some more, or should i merge? thanks!

**Comment by lgebhardt** [comment]
@mrose17 I'm done

**Pull request merged**